examples/applications/cloud-app/src/Program.cs

```diff
@@ -2,109 +2,482 @@
 // Licensed under the MIT license.
 
 using System;
+using System.CommandLine.Parsing;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
 using System.IO;
+using System.Linq;
+using System.Text;
 using System.Text.Json;
 using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
+using CarBridgeCloudApp;
+using Chariott.Common.V1;
+using Chariott.Runtime.V1;
+using Chariott.Streaming.V1;
+using DocoptNet;
 using MQTTnet;
-using MQTTnet.Client;
 using MQTTnet.Formatter;
 
 try
 {
-    var mqttFactory = new MqttFactory();
+    return await ProgramArguments.ParseToMain(args, Main);
+}
+catch (Exception ex)
+{
+    Console.Error.WriteLine(ex);
+    return 1;
+}
 
+static async Task<int> Main(ProgramArguments args)
+{
+    var jsonSerializerOptions = new JsonSerializerOptions { WriteIndented = true };
+    var utf8 = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+    var timeout = new Timeout(TimeSpan.FromSeconds(int.Parse(args.OptTimeout, NumberStyles.None, CultureInfo.InvariantCulture)));
+
+    var mqttFactory = new MqttFactory();
     using var mqttClient = mqttFactory.CreateMqttClient();
 
-    var mqttClientOptions =
-        new MqttClientOptionsBuilder()
-            .WithTcpServer(args is [var server, ..] ? server : "localhost")
-            .WithProtocolVersion(MqttProtocolVersion.V500)
-            .Build();
+    await timeout.ApplyAsync(cancellationToken =>
+    {
+        var options =
+            mqttFactory.CreateClientOptionsBuilder()
+                       .WithTcpServer(args.OptBroker)
+                       .WithProtocolVersion(MqttProtocolVersion.V500)
+                       .Build();
+        return mqttClient.ConnectAsync(options, cancellationToken);
+    });
 
-    await mqttClient.ConnectAsync(mqttClientOptions, CancellationToken.None);
     Console.Error.WriteLine("The MQTT client is connected.");
```

Comment:
[nit] should we write this to standard output instead?

Reply:
This is for information purposes only so it's like logging, which generally goes to standard error.